include/openthread/tcp.h

```diff
@@ -0,0 +1,616 @@
+ */
+otError otTcpReceiveByReference(otInstance *aInstance, otTcpEndpoint *aEndpoint, const otLinkedBuffer **aBuffer);
+
+/**
+ * Reorganizes the receive buffer to be entirely contiguous in memory.
+ *
+ * This is optional; an application can simply traverse the linked buffer
+ * chain obtained by calling @p otTcpReceiveByReference. Some
+ * applications may wish to call this function to make the receive buffer
+ * contiguous to simplify their data processing, but this comes at the expense
+ * of CPU time to reorganize the data in the receive buffer.
+ *
+ * @param[in]   aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]   aEndpoint  A pointer to the TCP endpoint structure representing the TCP endpoint on which to receive data.
+ *
+ * @retval OT_ERROR_NONE    Successfully completed the operation.
+ * @retval OT_ERROR_FAILED  Failed to complete the operation.
+ */
+otError otTcpReceiveContiguify(otInstance *aInstance, otTcpEndpoint *aEndpoint);
+
+/**
+ * Informs the TCP stack that the application has finished processing
+ * @p aNumBytes bytes of data at the start of the receive buffer and that the
+ * TCP stack need not continue maintaining those bytes in the receive buffer.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure representing the TCP endpoint on which to receive data.
+ * @param[in]  aNumBytes  The number of bytes consumed.
+ * @param[in]  aFlags     Flags specifying options for this operation (none yet).
+ *
+ * @retval OT_ERROR_NONE    Successfully completed the receive operation.
+ * @retval OT_ERROR_FAILED  Failed to complete the receive operation.
+ */
+otError otTcpCommitReceive(otInstance *aInstance, otTcpEndpoint *aEndpoint, size_t aNumBytes, uint32_t aFlags);
+
+/**
+ * Informs the connection peer that this TCP endpoint will not send more data,
+ * used when the application has no more data to send to the connection peer.
+ * For this connection, future reads on the connection peer will result in the
+ * "end of stream" condition, and future writes on this connection endpoint
+ * will fail.
+ *
+ * The "end of stream" condition only applies after any data previously
+ * provided to the TCP stack to send out has been received by the connection
+ * peer.
+ *
+ * @param[in]  aInstance  A pointer to the OpenThread instance to which the TCP endpoint belongs.
+ * @param[in]  aEndpoint  A pointer to the TCP endpoint structure representing the TCP endpoint to shut down.
+ *
+ * @retval OT_ERROR_NONE    Successfully queued the "end of stream" condition for transmission.
+ * @retval OT_ERROR_FAILED  Failed to queue the "end of stream" condition for transmission.
+ */
+otError otTcpSendEndOfStream(otInstance *aInstance, otTcpEndpoint *aEndpoint);
+
+/**
+ * Forcibly ends the TCP connection associated with this TCP endpoint. This
```

Comment:
nit: have brief in separate block above

Reply:
Done. I'm not 100% what you meant by "in separate block", can you please check that what I've done is acceptable?